File: web/safari.permissions.js
function requestDeviceMotionEventPermission(callback) {
    let requestPermission = DeviceMotionEvent.requestPermission;
    if (typeof requestPermission === "function") {
        requestPermission()
            .then(callback)
            .catch((_) => callback(null));
    }

    callback(null);
}

File: lib/safari_permissions.dart
@JS()
library;

import 'package:js/js.dart';

void requestDeviceMotionEventPermission(
  void Function(String?) callback,
) {
  _requestDeviceMotionEventPermission(allowInterop(callback));
}

@JS('requestDeviceMotionEventPermission')
external void _requestDeviceMotionEventPermission(
  void Function(String?) callback,
);
